go-controller/pkg/libovsdbops/template_var.go

```diff
@@ -0,0 +1,234 @@
+			Model: nbTemplate,
+			ModelPredicate: func(item *nbdb.ChassisTemplateVar) bool {
+				return item.Chassis == nbTemplate.Chassis
+			},
+			OnModelMutations: []interface{}{&nbTemplate.Variables},
+			ErrNotFound:      false,
+			BulkOp:           false,
+			InsertMutator:    mutator,
+		})
+	})
+
+	modelClient := newModelClient(nbClient)
+	return modelClient.CreateOrUpdateOps(ops, opModels...)
+}
+
+func CreateChassisTemplateVarOps(nbClient libovsdbclient.Client,
+	ops []libovsdb.Operation, templateVars ...TemplateMap) ([]libovsdb.Operation, error) {
+
+	return mutateChassisTemplateVarOps(nbClient, ovsdb.MutateOperationInsert,
+		ops, templateVars...)
+}
+
+func CreateOrUpdateChassisTemplateVarOps(nbClient libovsdbclient.Client,
+	ops []libovsdb.Operation, templateVars ...TemplateMap) ([]libovsdb.Operation, error) {
+
+	return mutateChassisTemplateVarOps(nbClient, MutateOperationUpdate,
+		ops, templateVars...)
+}
+
+func DeleteChassisTemplateVarOps(nbClient libovsdbclient.Client,
+	ops []libovsdb.Operation, templateVars ...TemplateMap) ([]libovsdb.Operation, error) {
+	opModels := []operationModel{}
+	forEachNBTemplateInMaps(templateVars, func(nbTemplate *nbdb.ChassisTemplateVar) {
+		deleteNbTemplate := &nbdb.ChassisTemplateVar{
+			Chassis:   nbTemplate.Chassis,
+			Variables: map[string]string{},
+		}
+		for name := range nbTemplate.Variables {
+			deleteNbTemplate.Variables[name] = ""
+		}
+		opModels = append(opModels, operationModel{
+			Model: deleteNbTemplate,
+			ModelPredicate: func(item *nbdb.ChassisTemplateVar) bool {
+				return item.Chassis == deleteNbTemplate.Chassis
+			},
+			OnModelMutations: []interface{}{&deleteNbTemplate.Variables},
+			ErrNotFound:      false,
+			BulkOp:           false,
+		})
+	})
+
+	modelClient := newModelClient(nbClient)
+	return modelClient.DeleteOps(ops, opModels...)
+}
+
+func CreateOrUpdateChassisTemplateVar(nbClient libovsdbclient.Client, templateVars ...TemplateMap) error {
+	ops, err := CreateOrUpdateChassisTemplateVarOps(nbClient, nil, templateVars...)
+	if err != nil {
+		return err
+	}
+	_, err = TransactAndCheck(nbClient, ops)
+	return err
+}
+
+func DeleteChassisTemplateVarByName(nbClient libovsdbclient.Client, templateVarNames []string) error {
+	deleteTemplateVar := &nbdb.ChassisTemplateVar{
+		Variables: make(map[string]string, len(templateVarNames)),
+	}
+	for _, templateName := range templateVarNames {
+		deleteTemplateVar.Variables[templateName] = ""
+	}
+	opModel := operationModel{
+		Model: deleteTemplateVar,
+		ModelPredicate: func(item *nbdb.ChassisTemplateVar) bool {
+			for _, templateName := range templateVarNames {
+				if _, found := item.Variables[templateName]; found {
+					return true
+				}
+			}
+			return false
+		},
+		OnModelMutations: []interface{}{&deleteTemplateVar.Variables},
+		ErrNotFound:      false,
+		BulkOp:           true,
+	}
+
+	m := newModelClient(nbClient)
+	return m.Delete([]operationModel{opModel}...)
```

Comment:
no need to make array and expand it. Just do:
`m.Delete(opModel)`

Reply:
This shows my golang skills are still at "copy from here and paste it there"-level. :)  Thanks for pointing it out!